openff/bespokefit/executor/services/coordinator/stages.py

```diff
@@ -530,6 +533,20 @@ async def _inject_bespoke_qc_data(
 
             target.reference_data = local_qc_data
 
+            from qcelemental.molutil import guess_connectivity
+
+            # Check that connectivity still matches the input molecule
+            original_mol: Molecule = Molecule.from_smiles(input_schema.smiles)
+            for qc_record in local_qc_data.qc_records:
+                for name, mol in qc_record.final_molecules.items():
+                    mol = Molecule.from_qcschema(mol)
```

Comment:
Interesting. I thought I got it to work as expected on a positive and negative test case (both with only one fragment). I remember being very surprised that it did what I expected 😅 I'll write a test and double check tomorrow.

I initially tried to use `ConnectivityFilter`, but I couldn't find a place where I could apply it as widely as here - I found one place that used a result collection but was concerned that it would have to be re-implemented every time someone added a qcgeneration type. I'll try `guess_connectivity` here again.

I raised an issue about writing a developer's guide because I was finding it really hard to figure out what part of the code I should be editing to do what I want.

Reply:
Yeah I think we might have to implement something similar here rather than use qcsubmit directly and putting it here should mean it works for all qc data types in future as they should all return a molecule and qcschema which we can compare. 

Yeah, great idea about the developer guide ill have to refresh myself 😅  and get back to your points!